sdk/cosmos/azure-cosmos-kafka-connect/src/main/java/com/azure/cosmos/kafka/connect/implementation/source/MetadataMonitorThread.java

```diff
@@ -175,56 +186,60 @@ private Mono<Boolean> shouldRequestTaskReconfigurationOnFeedRanges(List<CosmosCo
     }
 
     private Mono<Boolean> shouldRequestTaskReconfigurationOnFeedRanges(CosmosContainerProperties containerProperties) {
-        if (feedRangesMetadataOffsetExists(containerProperties)) {
-            CosmosAsyncContainer container =
-                this.cosmosClient
-                    .getDatabase(this.sourceContainersConfig.getDatabaseName())
-                    .getContainer(containerProperties.getId());
-
-            return container
-                .getFeedRanges()
-                .flatMap(feedRanges -> {
-                    FeedRangesMetadataTopicOffset topicOffset =
-                        this.offsetStorageReader
-                            .getFeedRangesMetadataOffset(
-                                this.sourceContainersConfig.getDatabaseName(),
-                                containerProperties.getResourceId());
-
-                    if (topicOffset == null) {
-                        // the container may have recreated
-                        return Mono.just(true);
-                    }
-
-                    List<FeedRange> differences =
-                        topicOffset
-                            .getFeedRanges()
-                            .stream()
-                            .filter(normalizedFeedRange -> !feedRanges.contains(normalizedFeedRange))
-                            .collect(Collectors.toList());
-
-                    if (differences.size() == 0) {
-                        // the feedRanges are exact the same
-                        return Mono.just(false);
-                    }
-
-                    // There are feedRanges change, but not all changes need to trigger a reconfiguration
-                    // Merge should not trigger task reconfiguration as we will continue pulling the data from the pre-merge feed ranges
-                    // Split should trigger task reconfiguration for load-balancing
-                    return shouldRequestTaskReconfigurationOnFeedRangeChanges(containerProperties, differences);
-                });
-        }
+        return feedRangesMetadataOffsetExists(containerProperties)
+            .flatMap(offsetExists -> {
+                if (offsetExists) {
+                    CosmosAsyncContainer container =
+                        this.cosmosClient
+                            .getDatabase(this.sourceContainersConfig.getDatabaseName())
+                            .getContainer(containerProperties.getId());
+
+                    return container
+                        .getFeedRanges()
```

Comment:
Use the one from cache instead?

Reply:
we will need to get the latest feedRanges from source to compare with what exists in the cache